### DISCUSSION TITLE
Why do GuardClauses return a value and not the `GuardClause` or `void`?

### DISCUSSION THREAD

**Discussion by Marvin-Brouwer** [ardalis/GuardClauses#244] [Q&A] [answered]
Hi I was just wondering why all the GuardClauses return the original value they were validating?
I tried using them to chain into a property but it just doesn't look right:

![image](https://user-images.githubusercontent.com/5499778/200259424-35e73ebd-a4e9-4aca-a39a-91c9fce504fc.png)
![image](https://user-images.githubusercontent.com/5499778/200259459-66a3fec5-cd1c-46f1-82c4-ca8377984dcc.png)

If I compare these approaches, the latter looks more readable to me.
So I'm just wondering what the usecase for the return value might be.

**Comment by amal-stack** [comment, marked as answer]
The purpose of the return value is to simplify value assignment in constructors. 
```c#
class Person
{
    public string Name { get; }
    public string Age { get; }
    public Person(string name, int age)
    {
        // return values used for assignment to Name and Age
         Name = Guard.Against.NullOrWhiteSpace(name); 
         Age = Guard.Against.NegativeOrZero(age);
    }
}
``` 
Without the return value, the constructor would be of four lines of code instead of two.
This has already been answered in #158.

**Comment by Marvin-Brouwer** [comment]
Aha, I see.
Maybe it violates the single responsibility principle a bit, but they're guidelines anyway.

Thanks.

**Comment by ardalis** [comment]
@amal-stack has the answer. It's purely for pragmatic reasons to make constructor assignment simpler.
